superset/db_engine_specs/redshift.py

```diff
@@ -148,7 +151,20 @@ def _mutate_label(label: str) -> str:
         return label.lower()
 
     @classmethod
-    def get_cancel_query_id(cls, cursor: Any, query: Query) -> Optional[str]:
+    def convert_dttm(
+        cls, target_type: str, dttm: datetime, db_extra: dict[str, Any] | None = None
+    ) -> str | None:
+        sqla_type = cls.get_sqla_column_type(target_type)
+
+        if isinstance(sqla_type, Date):
+            return f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
+        if isinstance(sqla_type, DateTime):
+            dttm_formatted = dttm.isoformat(sep=" ", timespec="microseconds")
+            return f"""TO_TIMESTAMP('{dttm_formatted}', 'YYYY-MM-DD HH24:MI:SS.US')"""
+        return None
```

Comment:
It hit me, we should consider moving this method from `PostgresEngineSpec` to `PostgresBaseEngineSpec`, as I think it's probably a good default for Postgres-like databases. This way the Redshift spec would get it "for free".

Reply:
+1

Reply:
yes sure, doing that

Reply:
done @john-bodley @villebro , redshift test is also running, i have ran `tox -e pre-commit` as well @john-bodley , please approve CI

Reply:
@john-bodley  i think ci is not started yet, can you help